src/js/pouch-express.js

```diff
@@ -0,0 +1,287 @@
+    }
+
+    // There are unfortunately options that can only be configured via a configuration file.
+    //
+    // To allow ourselves (and users configuring and extending this grade) to control these options, we create the file
+    // with the contents of options.pouchConfig before configuring and starting express-pouchdb.
+    fs.writeFileSync(that.options.expressPouchConfigPath, JSON.stringify(that.options.expressPouchConfig, null, 2));
+
+    that.PouchDB = PouchDB.defaults(that.options.dbOptions);
+    that.expressPouchdb = expressPouchdb(that.PouchDB, { configPath: that.options.expressPouchConfigPath });
+
+    return that.expressPouchdb;
+};
+
+/**
+ *
+ * Initialize all of the configured databases in `that.options.database`.
+ *
+ * @param that
+ * @returns {Promise} - A `fluid.promise.sequence` that will be resolved when all databases are initialized.
+ *
+ */
+gpii.pouch.express.initDbs = function (that) {
+    // We create our components programatically because we need:
+    //
+    //   1. To be notified when all databases are ready for use.  We do this with a `fluid.promise.sequence`.
+    //   2. A way to clean up each database later on.  We do this by retaining a list of database instances.
+    //   3. A way to know when all of them have been destroyed.  We do this with another `sequence` in our cleanup invoker (see below).
+    //
+    // I can think of ways to accomplish #2 with dynamic components, but not 1 and 3.
+    // TODO: Review with Antranig.
+
+    var promises = [];
+    fluid.each(that.options.databases, function (dbOptions, dbKey) {
+        promises.push(gpii.pouch.express.initDb(that, dbKey, dbOptions));
+    });
+
+    var sequence = fluid.promise.sequence(promises);
+    sequence.then(function () {
+        that.events.onStarted.fire();
+    });
+
+    return sequence;
+};
+
+/**
+ *
+ * Initialize a single database instance.
+ *
+ * @param that - The component itself
+ * @param dbKey {String} - The database name.
+ * @param dbDef {Object} - Our convention for representing multiple databases.  See the docs for examples.
+ * @returns {Promise} A promise that will be resolved with the database is initialized.
+ */
+gpii.pouch.express.initDb = function (that, dbKey, dbDef) {
+    var expandedDef = gpii.pouch.express.expandDbDef(dbDef);
+    var initPromise = fluid.promise();
+    var dbOptions = expandedDef.dbOptions ? fluid.merge(that.options.dbOptions, expandedDef.dbOptions) : fluid.copy(that.options.dbOptions);
+    dbOptions.name = dbKey;
+    var dbComponentOptions = {
+        gradeNames: that.options.pouchGradeNames,
+        dbOptions: dbOptions,
+        baseDir: that.options.baseDir,
+        listeners: {
+            "onDataLoaded.resolvePromise": {
+                func: initPromise.resolve
+            }
+        }
+    };
+    if (expandedDef.data) {
+        dbComponentOptions.dbPaths = expandedDef.data;
+    }
+
+    var dbComponent = fluid.component(dbComponentOptions);
```

Comment:
The programmatic API was used to support the previous `databases` option rather than forcing everyone to define subcomponents for each database.  I will try the Nexus methods here and we can review later.